core/src/main/java/hudson/PluginManager.java

```diff
@@ -778,45 +945,115 @@ public HttpResponse doUpdateSources(StaplerRequest req) throws IOException {
      * Performs the installation of the plugins.
      */
     public void doInstall(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
-        boolean dynamicLoad = req.getParameter("dynamicLoad")!=null;
+        Set<String> plugins = new LinkedHashSet<>();
 
         Enumeration<String> en = req.getParameterNames();
         while (en.hasMoreElements()) {
             String n =  en.nextElement();
             if(n.startsWith("plugin.")) {
                 n = n.substring(7);
-                // JENKINS-22080 plugin names can contain '.' as could (according to rumour) update sites
-                int index = n.indexOf('.');
-                UpdateSite.Plugin p = null;
+                plugins.add(n);
+            }
+        }
+
+        boolean dynamicLoad = req.getParameter("dynamicLoad")!=null;
+        install(plugins, dynamicLoad);
+
+        rsp.sendRedirect("../updateCenter/");
+    }
+
+    /**
+     * Installs a list of plugins from a JSON POST.
+     * @param req The request object.
+     * @return A JSON response that includes a "correlationId" in the "data" element.
+     * That "correlationId" can then be used in calls to
+     * {@link UpdateCenter#doInstallStatus(org.kohsuke.stapler.StaplerRequest)}.
+     * @throws IOException Error reading JSON payload fro request.
+     * @since FIXME
+     */
+    @RequirePOST
+    public JSONObjectResponse doInstallPlugins(StaplerRequest req) throws IOException {
```

Comment:
Why? This is called by the install wizard, which I assume would be considered to be "external". Other Jenkins distros will also want to call this and they'd def be "external". Why do `@Restricted` for this method and not do it for the other `install` method endpoint (called by the form post action in the PM)?